### PR TITLE
Allow calico to be removed from containers even if they are missing f…

### LINKED ISSUE #501: Can't remove a container from Calico if it's already been killed and rmed from Docker
I wanted to stop a Calico node (`calicoctl node stop`), but it complained (rightly) that there was still a workload running.  I `docker kill`ed and `docker rm`ed the container, but it remained in calicoctl.

I then tried `calicoctl container remove <name>`, `calicoctl container remove <short id from docker PS>` and `calicoctl container remove <long workload id from calicoctl endpoint show --detailed>`, but in all cases it returned "Container <identifier> was not found.".

I was using calico-docker 0.6.0, having followed the Ubuntu manual install process.  I was able to `calicoctl node stop --force`, but it should be possible to remove a container from Calico without doing this.

### PR DESCRIPTION
**Pull request by tomdee**
…rom docker.

Fixes #501

### CODE REVIEW

File: build_calicoctl/requirements.txt [hunk 1]
```diff
@@ -15,4 +15,4 @@ flask
 gunicorn
 subprocess32
 git+https://github.com/projectcalico/python-etcd.git
-git+https://github.com/projectcalico/libcalico.git
+git+https://github.com/tomdee/libcalico.git@namespace_path
```
Comment (insequent) [inline build_calicoctl/requirements.txt]: I'm assuming this is temporary?

**Comment by insequent** [comment]
Huge :+1:  to this! A very desired feature with my work.

File: build_calicoctl/requirements.txt (hunk 1 above)
Comment (tomdee) [inline build_calicoctl/requirements.txt]: Yes

**Comment by tomdee** [comment]
I'm planning to rework this slightly to remove the need for the workload_exists method.

**Comment by tomdee** [comment]
@robbrockbank or @spikecurtis Could one of you take a look at this? I've reworked it to minimize etcd reads and fix up a few bugs in the process.

**Comment by spikecurtis** [comment]
I'll take a look on Tuesday.  Sound good?

**Comment by tomdee** [comment]
@spikecurtis Are you going to be able to look at this today? Otherwise @robbrockbank do you have time this afternoon?

**Comment by robbrockbank** [comment]
@tomdee I will have time if spike doesn't

**Comment by tomdee** [comment]
Spike's out of the office this afternoon so can you take a look?

File: calico_containers/calico_ctl/container.py [hunk 2]
```diff
@@ -114,18 +115,6 @@ def container(arguments):
...
-            if container_id.startswith("/") and os.path.exists(container_id):
-                # The ID is a path. Don't do any docker lookups
-                workload_id = escape_etcd(container_id)
-                orchestrator_id = NAMESPACE_ORCHESTRATOR_ID
-            else:
-                info = get_container_info_or_exit(container_id)
-                workload_id = info["Id"]
-                orchestrator_id = DOCKER_ORCHESTRATOR_ID
-
         if arguments.get("ip"):
```
Comment (robbrockbank) [inline calico_containers/calico_ctl/container.py]: Shouldn't we be using the workload lookup here too?

File: calico_containers/calico_ctl/container.py [hunk 3]
```diff
@@ -300,47 +311,46 @@ def container_remove(container_id):
...
-    # Remove the container from the datastore.
-    client.remove_workload(hostname, orchestrator_id, workload_id)
-
-    print "Removed Calico interface from %s" % container_id
+        try:
+            # Remove the interface if it exists
+            netns.remove_veth(endpoint.name)
+        except CalledProcessError:
+            print "Could not remove Calico interface %s" % endpoint.name
+    else:
```
Comment (robbrockbank) [inline calico_containers/calico_ctl/container.py]: you never break from this loop, so this else is somewhat superfluous no?

File: calico_containers/tests/unit/container_test.py [hunk 4]
```diff
@@ -252,87 +252,88 @@ def test_container_remove(self, m_netns, m_client,  m_get_container_id,
...
-        m_get_container_id.assert_called_once_with("container1")
-        m_client.get_endpoint.assert_called_once_with(
-                                         hostname=utils.hostname,
-                                         orchestrator_id=utils.DOCKER_ORCHESTRATOR_ID,
-                                         workload_id=52)
-        m_client.get_ip_pools.assert_called_once_with(4)
-        self.assertFalse(m_client.remove_workload.called)
+            utils.hostname, utils.DOCKER_ORCHESTRATOR_ID, "container1")
+
+    # @patch('calico_ctl.container.enforce_root', autospec=True)
```
Comment (robbrockbank) [inline calico_containers/tests/unit/container_test.py]: any plan to fix these rather than comment them out?

**Comment by robbrockbank** [comment]
Few comments in line and need to investigate build failure.

File: calico_containers/calico_ctl/container.py (hunk 2 above)
Comment (tomdee) [inline calico_containers/calico_ctl/container.py]: Do you mean for the ip commands?

Comment (robbrockbank) [inline calico_containers/calico_ctl/container.py]: yup i mean the IP commands.  just nice to commonize the code when we get the chance.

i'm not too fussed if you want to just get this in.

File: calico_containers/calico_ctl/container.py (hunk 3 above)
Comment (tomdee) [inline calico_containers/calico_ctl/container.py]: Ah yes.

**Comment by tomdee** [comment]
This should be good to go now. @robbrockbank

**Comment by robbrockbank** [comment]
LGTM

**Pull request merged**